src/librustc_session/options.rs

```diff
@@ -499,6 +501,15 @@ macro_rules! options {
             }
         }
 
+        fn parse_cfguard(slote: &mut Option<CFGuard>, v: Option<&str>) -> bool {
+            match v {
+                Some("nochecks") => *slote = Some(CFGuard::Nochecks),
+                Some("checks") => *slote = Some(CFGuard::Checks),
```

Comment:
Why not make this a boolean option like `-Cno-redzone`?

Reply:
I don't think we can make this a boolean option because we need to encode three possibilities:
- option not present => do nothing
- `control_flow_guard=nochecks` => emit CFG metadata but no checks (corresponds to the Clang option `-cfguard,nochecks`). This is sometimes needed when linking against other binaries that expect CFG metadata.
- `control_flow_guard=checks` => emit CFG metadata and checks (corresponds to the Clang option `-cfguard`)

Reply:
Ah, I see, thanks for explaining!